TypeCobol/Compiler/Directives/TypeCobolOptions.cs

```diff
@@ -33,6 +33,15 @@ public bool UseEuroInformationLegacyReplacingSyntax
             set { _useEuroInformationLegacyReplacingSyntax = value; }
         }
 
+        /// <summary>
+        /// Are these Options used for parsing a Copy.
+        /// </summary>
+        public bool AreForCopyParsing
```

Comment:
Where this option is used is so far away, and will add such parameter to many methods.

Reply:
I agree too that AreForCopyParsing is not really an option.
When you parse a copy, you don't have the choice to parse it as a program.
It would better to store this information in something more technical / not part of public API.

For TextSourceInfo, I don't know. Because when this object is instantiated, you don't know if it's a copy yet.

Other solutions:
Scanner : Add a property? Convert static method to instance methods ?
TypeCobolOptions : at least change the visibility to internal and make it a read-only property.

Reply:
> For TextSourceInfo, I don't know. Because when this object is instantiated, you don't know if it's a copy yet.

Checking the usages show that the TextSourceInfo constructor is (indirectly) called by Workspace and FileCompiler. In Workspace, we can safely assume that it is a program, in FileCompiler we have the `isCopyFile` info. Rest of usages are in tests.

I agree that the internal + readonly could be a quick workaround.

Reply:
@smedilol , @fm-117 Ok what about using the InitialScanStateForCopy and add new Flag InsideCopy to MultilineScanState. and to propagate this flag.

Reply:
@mayanje @fm-117 Yes, it seems a good option. And if solution listed in #428 it will also works with new method Scanner. ScanIsolatedTokenInGivenContext.

Reply:
done